core/src/com/unciv/logic/civilization/QuestManager.kt

```diff
@@ -339,6 +437,66 @@ class QuestManager {
 
         return null
     }
+
+    /**
+     * Returns a Great Person [BaseUnit] that is not owned by both the [challenger] and the [civInfo]
+     */
+    private fun getGreatPersonForQuest(challenger: CivilizationInfo): BaseUnit? {
+        val ruleSet = civInfo.gameInfo.ruleSet
+        val greatPeople = ruleSet.units.values
```

Comment:
Disagree with the implementation. 
We should start with the possible great people - that is, great people that the challenger can get ( challenger.getCivGreatPeople() )
Then we only need to subtract civInfo.getCivGreatPeople()
All the 'civInfo.getEquivalentUnit' stuff can be done in the getCivGreatPeople() function itself.

Reply:
Sorry @yairm210 , I don't think I got how you plan to implement this. `getCivGreatPeople`, as I implemented it, returns a list of alive Great People (`MapUnit`s). 
Also, `getThisOrReplacedBaseUnit` (misleading name maybe) does the opposite of `getEquivalentUnit` (also misleading name imho): it takes a `BaseUnit` and returns its _primitive_, replaced, `BaseUnit`, if any. 

I am summarizing how I understood you'd like this to work, correct me if I'm wrong:
* get a list of _primitive_ spawnable great people ("Great General", "Great Scientist", etc)
* subtract alive great people for city state (no conversion required, i assume? maybe someone wants to mod city state to have unique units, so it'd be safer to assume they can have unique units?) 
* subtract alive great people for challenger, converting their unique units to their respective primitive base unit (this operation may be made by a function `BaseUnit.getReplacedUnit`